src/Exceptions/IllegalArgumentException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace RavenDB\Exceptions;
+
+use Exception;
+use Throwable;
+
+class IllegalArgumentException extends Exception implements Throwable
```

Comment:
I think you should use php build in type here: https://www.php.net/manual/en/class.invalidargumentexception.php